sympy/matrices/common.py

```diff
@@ -2158,6 +2158,9 @@ class _MinimalMatrix(object):
     def _new(cls, *args, **kwargs):
         return cls(*args, **kwargs)
 
+    def diff(self, *args):
```

Comment:
I don't think this is appropriate here. `_MinimalMatrix` has no `applyfunc` method (which it shouldn't because it's minimal). If you need a `_MinimalMatrix` with a `diff` for some reason, `_MinimalMatrix` should be subclassed.

Reply:
@siefkenj this is what is causing problems in this PR:
```
In [49]: m = CalculusOnlyMatrix(2, 1, [x, y])

In [50]: sympify(m)
```
Input number 50 raises an exception.

Reply:
When you `sympify` a matrix, it turns it into an `ImmutableMatrix`. There is no immutable version of the `CalculusOnlyMatrix`, so one should never make a call to `sympify(CalculusOnlyMatrix(2, 1, [x, y]))`

Reply:
I see. So we need to move `.diff( )` to `CalculusOnlyMatrix`? There's a test where a matrix of that class gets tested for a derivative.

Reply:
Yes, adding `diff` to `CalculusOnlyMatrix` would be appropriate.

Reply:
I've XFAILed the `CalculusOnlyMatrix.diff` test. Either way, I believe `.diff( )` shouldn't be defined on the objects, as SymPy provides various ways to call the derivative:

- `a.diff(b)`
- `diff(a, b)`
- `Derivative(a, b).doit()`
- `Derivative(a, b, evaluate=True)`

They should behave the same way. Defining `diff` in a class only defines the first call.